docs/adding_timing_metadata.md

```diff
@@ -0,0 +1,110 @@
+## Problem
+
+When adding new tests to the CTS you may occasionally see an error like this
+when running `npm test` or `npm run wpt`
+
+```
+ERROR: Tests missing from listing_meta.json. Please add the new tests (set subcaseMS to 0 if you cannot estimate it):
+  webgpu:shader,execution,expression,binary,af_matrix_addition:matrix:*
+
+/home/runner/work/cts/cts/src/common/util/util.ts:38
+    throw new Error(msg && (typeof msg === 'string' ? msg : msg()));
+          ^
+Error: 
+    at assert (/home/runner/work/cts/cts/src/common/util/util.ts:38:11)
+    at crawl (/home/runner/work/cts/cts/src/common/tools/crawl.ts:155:11)
+Warning: non-zero exit code 1
+ Use --force to continue.
+
+Aborted due to warnings.
+```
+
+What this error message is trying to tell us, is that there is no entry for
+`webgpu:shader,execution,expression,binary,af_matrix_addition:matrix:*` in
+`src/webgpu/listing_meta.json`.
+
+These entries are estimates for the amount of time that subcases take to run,
+and are used as inputs into the WPT tooling to attempt to portion out tests into
+approximately same sized chunks.
+
+If a value has been defaulted to 0 by someone, you will see warnings like this
+```
+...
+WARNING: subcaseMS≤0 found in listing_meta.json (allowed, but try to avoid):
+  webgpu:shader,execution,expression,binary,af_matrix_addition:matrix:*
+...
+```
+
+These messages should be resolved by adding appropriate entries to the JSON
+file.
+
+## Solution
+
+There exists tooling in the CTS repo for generating appropriate estimates for
+these values, though they do require some manual intervention. The rest of this
+doc will be a walkthrough of running these tools.
+
+### Default Value
+
+The first step is to add a default value for entry to 
+`src/webgpu/listing_meta.json`, since there is a chicken-and-egg problem for 
+updating these values.
+
+```
+  "webgpu:shader,execution,expression,binary,af_matrix_addition:matrix:*": { "subcaseMS": 0 },
+```
+
+(It should have a value of 0, since later tooling updates the value if the newer
+value is higher)
+
+### Websocket Logger
+
+The first tool that needs to be run is `websocket-logger`, which uses a side
+channel from WPT to report timing data when CTS is run via a websocket. This
+should be run in a separate process/terminal, since it needs to stay running
+throughout the following steps.
+
+At `tools/websocket-logger/`
+```
+npm install
```

Comment:
npm ci?

Reply:
Checked this again, cause I always get confused on npm ci vs npm i :upside_down_face:

The README.md for websocket-logger uses `npm install`, and there isn't a `package-lock.json` in that directory, so `npm ci` won't work by default, npm i needs to be run first.

I am not 100% sure that this is the correct way for things to be setup, but it seems to work to me.

@kainino0x wrote this I believe, but is currently away.
@austinEng thoughts?

Reply:
I think this is a situation to file an issue about it if you feel strongly and we can discuss whether or not to change it there. Since this PR is just meant to document how the existing code operates.

Reply:
IIUC `npm ci` installs exact versions. `npm i` installs *compatible* versions. So the package.json says v1.2.3 and you might get v1.7.9 since by semver rules that's compatible (even though it might not be compatible - bugs, and it might have a virus/trojans that people sneak into newer versions)

Reply:
Good point, I have filed https://github.com/gpuweb/cts/pull/2942 and will take a look.